File: test/testDWState.js
var process = require('process');
var fs = require('fs');
var assert = require('assert');

var datawire_connect = require('datawire_connect_1_0_0').datawire_connect;
var DatawireState = datawire_connect.state.DatawireState;

var args = process.argv.splice(process.execArgv.length + 2);

var goldInfo = fs.readFileSync(args[0]).toString().split("\n");

var i = 0;

var goldDefaultPath = goldInfo[i++];
var goldOrgID = goldInfo[i++];
var goldEmail = goldInfo[i++];
var goldTokens = {};
var services = [];

while (i < goldInfo.length) {
	if (goldInfo[i].length > 0) {
		var tokenInfo = goldInfo[i].split(':::');

		var svcHandle = tokenInfo[0];
		var svcToken = tokenInfo[1];

		goldTokens[svcHandle] = svcToken;
		services.push(svcHandle);
	}

	i++;
}

var dwState = new DatawireState();

assert.equal(dwState.defaultStatePath(), goldDefaultPath);

dwState = DatawireState.defaultState();

assert.equal(dwState.getCurrentOrgID(), goldOrgID);
assert.equal(dwState.getCurrentEmail(), goldEmail);

var delSvc = [];

for (var i = 0; i < services.length; i++) {
	var svcHandle = services[i];
	var svcToken = dwState.getCurrentServiceToken(svcHandle);

	assert.equal(svcToken, goldTokens[svcHandle]);
	delSvc.push(svcHandle);
}

for (var i = 0; i < delSvc.length; i++) {
	delete(goldTokens[delSvc[i]]);
}

assert.equal(Object.keys(goldTokens).length, 0);

File: test/testDWState.py
#!python

import os

import json

from datawire_connect.state import DatawireState as qDWState
from datawire.utils.state import DataWireState as pDWState

def jsonify(obj):
  return json.dumps(obj, indent=4, separators=(',',':'), sort_keys=True)

class TestQuarkDWState (object):
  def test_defaultStatePath(self):
    qState = qDWState()
    pState = pDWState()

    mDefaultPath = os.path.join(os.path.expanduser('~'), '.datawire', 'datawire.json')
    pDefaultPath = pState.state_path

    assert(qState.defaultStatePath() == mDefaultPath)
    assert(pDefaultPath == mDefaultPath)

  def test_contents(self):
    pState = pDWState()

    pJSON = pState.toJSON()

    qState = qDWState()
    qContents = qState.defaultStateContents()
    qJSON = jsonify(json.loads(qContents))

    assert(pJSON == qJSON)

  def test_loadState(self):
    pState = pDWState()

    qState = qDWState.defaultState()

    assert(qState.getCurrentOrgID() == pState.currentOrgID())
    assert(qState.getCurrentEmail() == pState.currentOrg()['email'])

    pSvcs = pState.currentOrg()['service_tokens'].keys()
    qSvcs = qState.getCurrentServices()

    assert(jsonify(qSvcs) == jsonify(pSvcs))

    for svc in qSvcs:
      assert(qState.getCurrentServiceToken(svc) == pState.currentServiceToken(svc))

if __name__ == '__main__':
    qState = qDWState.defaultState()

    print("%s" % qState.defaultStatePath()) 
    print("%s" % qState.getCurrentOrgID())
    print("%s" % qState.getCurrentEmail())

    for svc in qState.getCurrentServices():
        print("%s:::%s" % (svc, qState.getCurrentServiceToken(svc)))
